### PR TITLE
feat: withdraw Erc20

### PR DESCRIPTION
**Pull request by peterpeterparker**
# Motivation

Provide support to withdraw ckErc20 -> Erc20.

# Changes

- Expose new feature `withdrawErc20` (similar to existing `withdrawEth`).
- Rework ckEth minter errors parsing to bubble the details instead of parsing it to string
- Implement errors for new withdrawal type

### CODE REVIEW

**Comment by dskloetd** [comment]
I'm rather busy at the moment. You might want to ask someone on the Gix team to review.

**Comment by peterpeterparker** [comment]
@lmuntaner can you review this PR?

File: packages/cketh/src/errors/minter.errors.ts [hunk 1]
```diff
@@ -1,46 +1,155 @@
...
+export class LedgerInsufficientFundsError<T> extends LedgerError<T> {}
 
 export const createWithdrawEthError = (
   Err: WithdrawalError,
-): MinterWithdrawalError => {
+): MinterGenericError => {
   if ("TemporarilyUnavailable" in Err) {
-    return new MinterTemporaryUnavailableError(Err.TemporarilyUnavailable);
+    return new MinterTemporaryUnavailableError({
+      details: Err.TemporarilyUnavailable,
```
Comment (lmuntaner) [inline packages/cketh/src/errors/minter.errors.ts:41]: No `msg` for these?

**Review by lmuntaner — APPROVED** [review]
LGTM! Thanks!

File: packages/cketh/src/errors/minter.errors.ts (hunk 1 above)
Comment (peterpeterparker) [reply packages/cketh/src/errors/minter.errors.ts]: Good idea 👍. Done.

Comment (lmuntaner) [reply packages/cketh/src/errors/minter.errors.ts]: Thanks!

**Pull request merged**